### ISSUE TITLE
AnimatorSet crash "Sorting went bad, the start event should always be at index 0"

### ISSUE THREAD

**Issue by kmadsen** [mapbox/mapbox-maps-android#1410]
### tl;dr

There is a crash and we don't know if it is a navigation issue, [an android animation framework issue](https://issuetracker.google.com/issues/234749373), or a maps issue. This issue is looking for potential causes or solutions from maps

### The crash

navigation: [v2.5.0-rc.1](https://github.com/mapbox/mapbox-navigation-android/releases/tag/v2.5.0-rc.1)
maps: [android-v10.5.0](https://github.com/mapbox/mapbox-maps-android/releases/tag/android-v10.5.0)

```
Caused by: java.lang.UnsupportedOperationException: Sorting went bad, the start event should always be at index 0
        at android.animation.AnimatorSet.sortAnimationEvents(AnimatorSet.java:1638)
        at android.animation.AnimatorSet.createDependencyGraph(AnimatorSet.java:1534)
E/AndroidRuntime:     at android.animation.AnimatorSet.initAnimation(AnimatorSet.java:700)
        at android.animation.AnimatorSet.start(AnimatorSet.java:719)
        at android.animation.AnimatorSet.start(AnimatorSet.java:684)
        at com.mapbox.navigation.ui.maps.camera.NavigationCamera.startAnimation(NavigationCamera.kt:418)
        at com.mapbox.navigation.ui.maps.camera.NavigationCamera.startAnimation$default(NavigationCamera.kt:399)
        at com.mapbox.navigation.ui.maps.camera.NavigationCamera.updateFrame(NavigationCamera.kt:338)
        at com.mapbox.navigation.ui.maps.camera.NavigationCamera.sourceUpdateObserver$lambda-1(NavigationCamera.kt:156)
        at com.mapbox.navigation.ui.maps.camera.NavigationCamera.$r8$lambda$q1UBn9Kis_Bd006m3L-ZIp9dt4A(Unknown Source:0)
        at com.mapbox.navigation.ui.maps.camera.NavigationCamera$$ExternalSyntheticLambda0.viewportDataSourceUpdated(Unknown Source:2)
        at com.mapbox.navigation.ui.maps.camera.data.MapboxNavigationViewportDataSource.setViewportData(MapboxNavigationViewportDataSource.kt:306)
        at com.mapbox.navigation.ui.maps.camera.data.MapboxNavigationViewportDataSource.evaluate(MapboxNavigationViewportDataSource.kt:381)
```

### Why could it be maps?

The `AnimatorSet` is built with a [set of maps animations](https://github.com/mapbox/mapbox-navigation-android/blob/77ca26c88be12571a86c4bef026bcd4b15492769/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/transition/MapboxNavigationCameraTransition.kt#L209-L276)

```
cameraPlugin.createCenterAnimator
cameraPlugin.createZoomAnimator
cameraPlugin.createBearingAnimator
cameraPlugin.createPitchAnimator
cameraPlugin.createPaddingAnimator
```

All the [`CameraAnimator` instances are custom `ValueAnimator`](https://github.com/mapbox/mapbox-maps-android/blob/6b0b4a0925f4655bab7c4bdc443a33343d65c881/plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/animator/CameraAnimator.kt#L24)

So I'm looking for a potential combination of these, that would trigger the "Sorting went bad, the start event should always be at index 0" crash. Is it possible for custom ValueAnimators to cause an incompatibility with `AnimatorSet`?

**Comment by kmadsen** [comment]
cc: @pengdev there is this bug fix recently that effects the codepath https://github.com/mapbox/mapbox-maps-android/pull/1345 

Would that bug be part of 10.5.0? Or is there something else you may see as a potential cause?

**Comment by yunikkk** [comment]
@kmadsen seems the error in the animator is thrown a https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/animation/AnimatorSet.java#1637

**Comment by pengdev** [comment]
> cc: @pengdev there is this bug fix recently that effects the codepath #1345 .. issue does not look related

I don't think the issue is related to #1345, as the animator owner change only affects how we handle gestures in the viewport plugin.

> Would that bug be part of 10.5.0? Or is there something else you may see as a potential cause?

Have you reproduced the issue with other versions of maps SDK? Or are there reproducible code snippets?

**Comment by kmadsen** [comment]
> @ kmadsen the error in the animator is thrown at 
> https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/animation/AnimatorSet.java#1637

Yeah, but this code is quite old and used in a lot of places. Even if it is an issue in the android framework, it will take a bit of time to get a result. But that thread is here https://issuetracker.google.com/issues/234749373. It is also odd that this crash is new for Mapbox.

> Have you reproduced the issue with other versions of maps SDK? Or are there reproducible code snippets?

This is part of the issue, I'm not sure how to reproduce it. I've been trying to reproduce it with directly with AnimatorSet but I can't find a combination of `startDelay` and `duration` that would cause it. The next theory is that it has something to do with an integration with custom ValueAnimators (CameraAnimator?). Also tried finding race conditions where, maybe the [`AnimatorSet.mEvents.clear()`](https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/animation/AnimatorSet.java#1541) is called before the crash check happens.

**Comment by kmadsen** [comment]
Closing as we cannot find a way to reproduce. Will re-open if we find something
